### ISSUE TITLE
[BUG] Error on `return` a `SIMD[ , 2]` in `alias`

### ISSUE THREAD

**Issue by rd4com** [modularml/mojo#3598]
### Bug description

Hello, it is a bug experienced by @soraros , on the chat:

```mojo
fn f() -> SIMD[DType.float32, 2]:
  return SIMD[DType.float32, 2](0, 1)

fn main():
  alias v = f()
  print(v)
```

> failed to fold operation `pop.simd.insertelement`

&nbsp;

It works great with:
- `return SIMD[DType.float32, 2](0)`
- ```mojo
  fn f() -> SIMD[DType.float32, 2] as output:
    output = SIMD[DType.float32, 2](0,1)
  
  fn main():
    alias v = f()
    print(v)
  ```
- ```mojo
  fn f() -> SIMD[DType.float32, 2]:
    s = SIMD[DType.float32, 2]()
    s[0] = 0 
    s[1] = 1
    return s
  
  fn main():
    alias v = f()
    print(v)
  ```
Maybe the overload `SIMD.__init__` with `*args` ?

### Steps to reproduce

Nothing special, it is in the description.

### System information

```shell
- `Ubuntu 23.10`
- `mojo 2024.10.205 (26f6391e)`
- `magic 0.3.0`
```

**Comment by rd4com** [comment]
In addition, @soraros provided the next example:
```mojo
fn main():
  alias b = f()
  print(b)

@always_inline
fn f() -> Bool:
  a = UInt(1 << 32)
  return a > 0

```
Does this one needs to be moved into a new GH issue ?

**Comment by soraros** [comment]
Both seem to be fixed on mojo 2024.10.605 (10ee785d).

ref: 40c2fffb30b0a6571f854340b70677a12ae7b955

**Comment by rd4com** [comment]
✅ Fixed by commit https://github.com/modularml/mojo/commit/40c2fffb30b0a6571f854340b70677a12ae7b955
